### PR TITLE
Update Prow config per the deprecation notice

### LINKED ISSUE #1812: Update Prow config files based on the deprecation notice
According to the announcement in https://github.com/kubernetes/test-infra/blob/master/prow/ANNOUNCEMENTS.md, there are two breaking changes in Prow that will come soon:
1. Added a default_decoration_configs option to the Plank config that allows to specify different plank's default configuration for each organization or a specific repository. default_decoration_config will be deprecated in April 2020 and it will be replaced with the * value in default_decoration_configs.
2. January 03, 2020 Added a pr_status_base_urls option to the Tide config that allows to specify different tide's URL for each organization or a specific repository. The pr_status_base_url will be deprecated on June 2020 and it will be replaces with the * value in pr_status_base_urls.

https://github.com/knative/test-infra/pull/1792 was supposed to fix them, but the `checkconfig` stops working as we haven't updated the bazel rules. We need to instead use the `checkconfig` image in a separate presubmit test job to check the config, before fixing the breaking changes. The issue is documented in https://github.com/knative/test-infra/issues/1810.

/assign

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
Update Prow config per the deprecation notice

**Which issue(s) this PR fixes**:
Fixes #1812

This one depends on other PRs, so I will do a rebase after this one is ready to be merged.
/hold
/cc @chaodaiG

### CODE REVIEW

File: config/prow-staging/core/config.yaml [hunk 1]
```diff
@@ -21,21 +21,22 @@ plank:
...
-      bucket: "knative-prow-staging"
-      path_strategy: "explicit"
-    gcs_credentials_secret: "test-account"
+  default_decoration_configs:
+    '*':
+      timeout: 2h
+      grace_period: 15s
+      utility_images:
+        # Update these versions when updating plank version in cluster.yaml
+        clonerefs: "gcr.io/k8s-prow/clonerefs:v20200225-55a0aacbd"
```
Comment (chaodaiG) [inline config/prow-staging/core/config.yaml:32]: why roll this back?

Comment (chizhg) [reply config/prow-staging/core/config.yaml]: Sorry, copy-paste error. Changed back.

**Pull request merged**